### PR TITLE
Add support Beacon UUID and iBeacon

### PR DESCRIPTION
**Pull request by dmamontov**
Quite a big improvement, adds the ability to track the indication of the Beacon UUID as a separate identifier, instead of MAC. The matter is that MAC in such devices can be dynamic.

Also added support for the iBeacon protocol.
Many more may be added in the future.

Below is an example of an android phone in which the home assistant emulates the iBeacon protocol:
image

### CODE REVIEW

File: custom_components/ble_monitor/__init__.py [hunk 1]
```diff
@@ -513,9 +522,9 @@ def __init__(self, config, dataqueue):
         if self.config[CONF_DEVICES]:
             for device in self.config[CONF_DEVICES]:
                 if CONF_DEVICE_ENCRYPTION_KEY in device and device[CONF_DEVICE_ENCRYPTION_KEY]:
-                    p_mac = bytes.fromhex(device["mac"].replace(":", "").lower())
+                    p_id = bytes.fromhex(dict_get_or_clean(device).lower())
```
Comment (Ernst79) [inline custom_components/ble_monitor/__init__.py:525]: Just that I understand it, can `p_id` either be a MAC address or an UUID?

Comment (dmamontov) [reply custom_components/ble_monitor/__init__.py:525]: everything is right. there are types of beacon with encryption. I want to add more of them, but for now I added ibeacon as an example

**Comment by Ernst79** [comment]
It looks good to me, I propose we make a beta release first, to test it.

**Comment by Ernst79** [comment]
Released as 7.3.0-beta

**Comment by dmamontov** [comment]
Fine. Thanks.

**Pull request merged**